From e1199b1fdcf09549cf415bb96a34d14633324719 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Amadeusz=20=C5=BBo=C5=82nowski?= <aidecoe@aidecoe.name>
Date: Wed, 25 Jul 2012 20:37:06 +0200
Subject: [PATCH 02/14] Makefile: use implicit rules for
 install/dracut-install

Don't fail on install rule if install/dracut-install is not build - just
skip it.  $(strip) is not used by install rule, because it is nowhere
set.

Conflicts:

	Makefile
---
 Makefile |   29 ++++++++++++++---------------
 1 files changed, 14 insertions(+), 15 deletions(-)

diff --git a/Makefile b/Makefile
index 0b29161..e36c2c2 100644
--- a/Makefile
+++ b/Makefile
@@ -8,6 +8,8 @@ pkglibdir ?= ${libdir}/dracut
 sysconfdir ?= ${prefix}/etc
 bindir ?= ${prefix}/bin
 mandir ?= ${prefix}/share/man
+CFLAGS ?= -O2 -g -Wall
+CFLAGS += -std=gnu99
 
 manpages = dracut.8 \
 	   dracut.cmdline.7 \
@@ -18,22 +20,17 @@ manpages = dracut.8 \
 
 .PHONY: install clean archive rpm testimage test all check AUTHORS doc
 
-all: syncheck dracut-version.sh dracut-install
+DRACUT_INSTALL_BIN = install/dracut-install
 
-DRACUT_INSTALL_SOURCE = \
-	install/dracut-install.c \
-	install/hashmap.c\
-	install/log.c \
-	install/util.c
+DRACUT_INSTALL_OBJECTS = \
+        install/dracut-install.o \
+        install/hashmap.o\
+        install/log.o \
+        install/util.o
 
-DRACUT_INSTALL_HEADER = \
-	install/hashmap.h \
-	install/log.h \
-	install/macro.h \
-	install/util.h
+all: syncheck dracut-version.sh $(DRACUT_INSTALL_BIN)
 
-dracut-install: $(DRACUT_INSTALL_SOURCE) $(DRACUT_INSTALL_HEADER)
-	gcc -std=gnu99 -O2 -g -Wall -o dracut-install $(DRACUT_INSTALL_SOURCE)
+$(DRACUT_INSTALL_BIN): $(DRACUT_INSTALL_OBJECTS)
 
 indent:
 	indent -i8 -nut -br -linux -l120 install/dracut-install.c
@@ -88,7 +85,9 @@ install: doc dracut-version.sh
 		ln -s ../dracut-shutdown.service \
 		$(DESTDIR)$(systemdsystemunitdir)/shutdown.target.wants/dracut-shutdown.service; \
 	fi
-	install $(strip) -m 0755 dracut-install $(DESTDIR)$(pkglibdir)/dracut-install
+	if [ -x $(DRACUT_INSTALL_BIN) ]; then \
+		install -m 0755 $(DRACUT_INSTALL_BIN) $(DESTDIR)$(pkglibdir)/dracut-install; \
+	fi
 
 dracut-version.sh:
 	@echo "DRACUT_VERSION=$(VERSION)-$(GITVERSION)" > dracut-version.sh
@@ -99,7 +98,7 @@ clean:
 	$(RM) */*/*~
 	$(RM) test-*.img
 	$(RM) dracut-*.rpm dracut-*.tar.bz2
-	$(RM) dracut-install
+	$(RM) $(DRACUT_INSTALL_BIN) $(DRACUT_INSTALL_OBJECTS)
 	$(RM) $(manpages) dracut.html
 	$(MAKE) -C test clean
 
-- 
1.7.8.6

